dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/tools/BeanMethodHandler.java

```diff
@@ -89,10 +95,16 @@ public BeanMethodHandler(Jsonb jsonb, BeanManager bm, MethodMetadata method) {
 
     protected Object[] constructArgsArray(ToolArguments t) {
         Object[] argsArray = new Object[method.argNames().length];
+        Map<String, Object> providedArgs = t.args();
         int i = 0;
         for (String name : method.argNames()) {
             if (name != null) {
-                argsArray[i] = t.args().get(name);
+                ArgumentMetadata argMetadata = arguments.get(name);
+                if (argMetadata != null) {
+                    argsArray[i] = ArgumentResolver.resolveArgumentValue(argMetadata, name, providedArgs);
+                } else {
+                    argsArray[i] = providedArgs.get(name);
+                }
```

Comment:
We should make `ToolArguments` contain the resolved argument values.

At the moment it doesn't make difference, but the interface for registering tools programatically allow a default value to be provided: https://github.com/OpenLiberty/open-liberty/blob/125258bd33f4ccddeeba09e58c28b1767886ea10/dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/tools/ToolManager.java#L93-L103

This only makes sense if the `ToolArguments` object provided to the handler already has these default values resolved.